packages/core/src/testers/testers.ts

```diff
@@ -47,15 +47,17 @@ import { deriveTypes, hasType, resolveSchema } from '../util';
 export const NOT_APPLICABLE = -1;
 /**
  * A tester is a function that receives an UI schema and a JSON schema and returns a boolean.
+ * Optionally, a root JSON schema can be provided, which will be used for resolving refs in the schema.
  */
-export type Tester = (uischema: UISchemaElement, schema: JsonSchema) => boolean;
+export type Tester = (uischema: UISchemaElement, schema: JsonSchema, rootSchema?: JsonSchema) => boolean;
```

Comment:
It doesn't need to be optional, but I made it so such that existing client code does not break. Would you prefer to have it as a required prop?

Reply:
Does it really break when it's not optional? Their existing testers will just ignore the additional parameter. Can you give an example which then breaks?

Reply:
When calling the testers directly, like will do in the tests, e.g. `schemaTypeIs('string')(uischema, schema, schema)`. But I guess this does not happen that often outside our code.

Reply:
I see. If it's fine for you I would like to see it made mandatory. When it's not provided it might be masking a bug. Within the code where we finally access the `rootSchema` we can stay defensive and still add a `rootSchema ?? schema` check to make sure we don't crash at runtime.